src/services/payment.service.ts

```diff
@@ -406,16 +449,50 @@ export class PaymentService {
       return {
         clientSecret: '',
         id: '',
+        type: 'fiat',
         invoiceStatus: finalizedInvoice.status,
       };
     }
 
+    const isLifetime = invoiceItem.price?.type === 'one_time';
+
+    if (isLifetime && this.bit2MeService.isAllowedCurrency(currency)) {
+      const invoice = await this.bit2MeService.createCryptoInvoice({
+        description: `Payment for lifetime product ${priceId}`,
+        priceAmount: invoiceItem.amount,
+        priceCurrency: currency,
+        title: `Invoice from Stripe ${finalizedInvoice.id}`,
+        securityToken: jwt.sign(
+          {
+            invoiceId: finalizedInvoice.id,
```

Comment:
Perhaps we could call it `stripeInvoiceId` to differentiate this invoice ID which is from Stripe, from the bit2me invoice ID.

Reply:
Agree.